common/src/gadgets/te_cond_add.rs

```diff
@@ -0,0 +1,320 @@
+use ark_ec::twisted_edwards::{Affine, TECurveConfig};
+use ark_ec::{AffineRepr, CurveGroup};
+use ark_ff::{FftField, Field};
+use ark_poly::univariate::DensePolynomial;
+use ark_poly::{Evaluations, GeneralEvaluationDomain};
+use ark_std::{vec, vec::Vec};
+use core::marker::PhantomData;
+
+use crate::domain::Domain;
+use crate::gadgets::booleanity::BitColumn;
+use crate::gadgets::cond_add::{AffineColumn, CondAdd, CondAddValues};
+use crate::gadgets::{ProverGadget, VerifierGadget};
+use crate::{const_evals, Column, FieldColumn};
+
+// Conditional affine addition:
+// if the bit is set for a point, add the point to the acc and store,
+// otherwise copy the acc value
+pub struct TeCondAdd<F: FftField, P: AffineRepr<BaseField = F>> {
+    pub(super) bitmask: BitColumn<F>,
+    pub(super) points: AffineColumn<F, P>,
+    // The polynomial `X - w^{n-1}` in the Lagrange basis
+    pub(super) not_last: FieldColumn<F>,
+    // Accumulates the (conditional) rolling sum of the points
+    pub acc: AffineColumn<F, P>,
+    pub result: P,
+}
+
+pub struct TeCondAddValues<F: Field, Curve: TECurveConfig<BaseField = F>> {
+    pub bitmask: F,
+    pub points: (F, F),
+    pub not_last: F,
+    pub acc: (F, F),
+    pub _curve: PhantomData<Curve>,
+}
+
+impl<F, Curve> CondAdd<F, Affine<Curve>> for TeCondAdd<F, Affine<Curve>>
+where
+    F: FftField,
+    Curve: TECurveConfig<BaseField = F>,
+{
+    type CondAddValT = TeCondAddValues<F, Curve>;
+    // Populates the acc column starting from the supplied seed (as 0 doesn't work with the addition formula).
+    // As the TE addition formula used is not complete, the seed must be selected in a way that would prevent
```

Comment:
Why is not complete? TE have complete addition formulas

Reply:
There are monomials in the denominator of the complete formula of degree four, that gets multiplied with selector polynomial (if the participant has signed or not) and the monomial x_3, results in a degree 6 which make the constrains of degree N=6n - c.

The monomials in incomplete formula are all of degree 2 which results in constrains of degree N=4n - c. 

Given FFT's comlpexity is N log N using complete formula would results in a significant slow dow  in the proof generation.